.github/workflows/pull_request.yaml

```diff
@@ -53,3 +53,55 @@ jobs:
           files: ./coverageunit.out
           verbose: true
           fail_ci_if_error: false
+
+  bench:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: './go.mod'
+
+      - id: go-cache-paths
+        run: |
+          echo "::set-output name=go-build::$(go env GOCACHE)"
+          echo "::set-output name=go-mod::$(go env GOMODCACHE)"
+
+      # Cache go build cache, used to speedup go test
+      - uses: actions/cache@v3
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-build }}
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+
+      # Cache go mod cache, used to speedup builds
+      - uses: actions/cache@v3
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-mod }}
+          key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}
+
+      # Run benchmark with `go test -bench` and stores the output to a file
+      - name: Run benchmark
+        run: go test ./... -bench=. -run=XXX | tee bench_output.txt
+
+      # Download previous benchmark result from cache (if exists)
+      - name: Download previous benchmark data
+        uses: actions/cache@v1
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          # What benchmark tool the output.txt came from
+          tool: 'go'
+          # Where the output from the benchmark tool is stored
+          output-file-path: bench_output.txt
+          # Where the previous data file is stored
+          external-data-json-path: ./cache/benchmark-data.json
```

Comment:
Shouldn't this be `./cache/bench_output.txt`?

Reply:
This file is different and contains the previous runs results. https://github.com/benchmark-action/github-action-benchmark#external-data-json-path-optional

I am kind of wondering how this works between branches? I think we may have to add a `main` workflow.

Reply:
Okay, updated to only compare on PRs and save on push to main.

Reply:
> This file is different and contains the previous runs results. https://github.com/benchmark-action/github-action-benchmark#external-data-json-path-optional

Right but in the `main` workflow you are saving the benchmark results to `output-file-path: bench_output.txt`... i don't understand where the json comes from 😕